cmd/juju/cloud/add.go

```diff
@@ -137,6 +148,14 @@ func (c *addCloudCommand) runInteractive(ctxt *cmd.Context) error {
 		return errors.Trace(err)
 	}
 
+	pollster.VerifyURLs = func(s string) (ok bool, msg string, err error) {
+		err = c.Ping(provider, s)
```

Comment:
Instead of always exceeding, let's do this:

```go
if errors.IsNotImplemented(err) == false {
    return false, "Can't validate endpoint: " + err.Error(), nil
}
```